extensions-api/src/main/scala/pl/touk/nussknacker/security/AuthCredentials.scala

```diff
@@ -1,3 +1,85 @@
 package pl.touk.nussknacker.security
 
-final case class AuthCredentials(value: String) extends AnyVal
+import java.nio.charset.StandardCharsets.UTF_8
+import java.util.Base64
+import javax.crypto.{Cipher, KeyGenerator, SecretKey}
+import scala.util.{Failure, Success, Try}
+
+sealed trait AuthCredentials
+
+object AuthCredentials {
+
+  val anonymousUsername     = "anonymous"
+  private val anonymousPass = "anonymous"
+
+  final case class PassedAuthCredentials(value: String) extends AuthCredentials
+  case object AnonymousAccess                           extends AuthCredentials
+
+  def fromStringToPassedAuthCredentials(value: String): AuthCredentials = {
+    PassedAuthCredentials(value)
+  }
+
+  def fromStringToAuthCredentials(value: String, crypter: Crypter): AuthCredentials = {
+    new String(Base64.getDecoder.decode(value)).split(":").toList match {
+      case `anonymousUsername` :: pass :: Nil =>
+        crypter.decrypt(pass) match {
+          case Failure(_) =>
+            PassedAuthCredentials(value)
+          case Success(decryptedAnonymousPassword) =>
+            decryptedAnonymousPassword match {
+              case `anonymousPass` => AnonymousAccess
+              case _               => PassedAuthCredentials(value)
+            }
+        }
+      case _ =>
+        PassedAuthCredentials(value)
+    }
+  }
+
+  implicit class Stringify(authCredentials: AuthCredentials) {
+
+    def stringify(crypter: Crypter): String = authCredentials match {
+      case PassedAuthCredentials(value) => value
+      case AnonymousAccess =>
+        new String(Base64.getEncoder.encode(s"$anonymousUsername:${crypter.encrypt(anonymousPass)}".getBytes))
```

Comment:
I wonder why this Auth credential need to look like this. Why we encode password and add the user instead of use some secure random and check if this secure random is this one that we generated. Can't it be just as simple? At the end we just want to check if our fake credentials are our own.

Reply:
not relevant anymore